javatests/arcs/core/delegates/DelegateTest.kt

```diff
@@ -0,0 +1,39 @@
+package arcs.core.delegates
+
+import arcs.BooleanDelegate
+import arcs.NumDelegate
+import arcs.TextDelegate
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+/** Tests for [StringDecoder]. */
+@Suppress("UNCHECKED_CAST", "UNUSED_VARIABLE")
+@RunWith(JUnit4::class)
+class DelegateTest {
+
+    @Test
+    fun TestNumDelegate() {
+        var n: Double by NumDelegate()
+        assertThat(n).isEqualTo(0.0)
+        n = 10.0
+        assertThat(n).isEqualTo(10.0)
+    }
```

Comment:
I feel like we should also be testing the `isSet` property. However it is private, and you generally shouldn't test private implementation details.

How is that property supposed to be used? It might be worth testing via some public API.

Reply:
`isSet` will eventually be used to enforce fields when new entities are written to handles (so at the wasm ABI boundary), which means it'll need to be accessible in some way. If Kotlin has some way of allowing access to some limited classes (like package private or C++ friending), that would be preferable, but read-only public would also be fine.

Reply:
Now at a completely code-generated solution so those tests went away.